### PR TITLE
Support HTTPS with client certificate authentication

### LINKED ISSUE #109: Support HTTPS with client certificate authentication.
We can simply terminate https with client certificate authentication by haproxy or stunnel. If you wish I can show this configuration for those application.

Such configurations can be next security level for private openhab installation.
We need support for https with client certificate authentication by android application.

On android system we can import CA and personal certificate to system certificate store.  It will be good if application can use this.
##

### PR DESCRIPTION
**Pull request by slawekjaranowski**
Fix #109

### CODE REVIEW

**Comment by digitaldan** [comment]
Awesome!  I need to get a release out this week, then I can get to reviewing this.  Sorry about missing this PR for so long!

File: KeyAuth.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# HTTPS with client certificate authentication
+
+To protect your openHAB instance in the public Internet network we can use client authenticate with SSL certificate feature.
+
+In order to prepare environment to work with this feature we must to do some below steps:
+
+## Own Certificate Authority
+TODO
```
Comment (digitaldan) [inline KeyAuth.md]: Do you have plans to fill this out?

Comment (slawekjaranowski) [inline KeyAuth.md]: Yes, I have such plans. I will try do it in next weekend.

I think that the best solutions is put links to forum and to documentation page, like:
https://community.openhab.org/t/using-nginx-reverse-proxy-authentication-and-https/14542 
http://docs.openhab.org/configuration/nginx.html

And next add information how to prepare and configure client certificate into those page.

Comment (digitaldan) [inline KeyAuth.md]: That would be great, I will go ahead and merge this and look forward to another PR for the readme!

**Pull request merged**